src/Core/StaticConfigTrait.php

```diff
@@ -247,17 +304,39 @@ public static function parseDsn($dsn)
     }
 
     /**
-     * Returns or updates the DSN class map for this class
+     * Updates the DSN class map for this class.
+     *
+     * @param array $map Additions/edits to the class map to apply.
+     * @return void
+     */
+    public static function setDsnClassMap(array $map)
```

Comment:
A DSN is the following:

```
scheme://user:password@host:port/path?query=args
```

We added support in Cake3 to use them for configuring datastores. You can now set something like:


```php
`Datasources.default.url` => `mysql://user:pass@localhost:3306/my_app`
```

To configure your default datasource to point at mysql.

The DSN class map maps DSN schemes to a cakephp class:

```
'file' => `Cake\Log\Engine\FileEngine`
'syslog' => `Cake\Log\Engine\SyslogEngine`
```

You can add extra classes to this map - like elasticsearch - at will where necessary. These are used when parsing configuration in your application's bootstrap. It's definitely configuration-related, though you could move it to a `DsnConfigTrait` if you really wanted to be anal about it.